iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ValidationUtil.java

```diff
@@ -256,25 +349,40 @@ public static void checkBPerSection(Validable validable, double sectionB) {
 
     public static void checkMaximumSectionCount(Validable validable, int maximumSectionCount) {
         if (maximumSectionCount <= 0) {
-            throw new ValidationException(validable,
-                    "the maximum number of section (" + maximumSectionCount
-                            + ") should be greater than 0");
+            throw new ValidationException(validable, "the maximum number of section (" + maximumSectionCount
+                    + ") should be greater than 0");
         }
     }
 
-    public static void checkSections(Validable validable, int currentSectionCount, int maximumSectionCount) {
-        if (currentSectionCount < 0) {
-            throw new ValidationException(validable,
-                    "the current number of section (" + currentSectionCount
-                            + ") should be greater than or equal to 0");
-        }
+    public static ValidationLevel checkSections(Validable validable, Integer currentSectionCount, int maximumSectionCount) {
+        return checkSections(validable, currentSectionCount, maximumSectionCount, true);
+    }
+
+    public static ValidationLevel checkSections(Validable validable, Integer currentSectionCount, int maximumSectionCount, boolean throwException) {
+        return checkSections(validable, currentSectionCount, maximumSectionCount, throwException, Reporter.NO_OP);
+    }
+
+    public static ValidationLevel checkSections(Validable validable, Integer currentSectionCount, int maximumSectionCount, boolean throwException, Reporter reporter) {
+        ValidationLevel validationLevel = ValidationLevel.LOADFLOW;
         checkMaximumSectionCount(validable, maximumSectionCount);
-        if (currentSectionCount > maximumSectionCount) {
-            throw new ValidationException(validable,
-                    "the current number (" + currentSectionCount
-                            + ") of section should be lesser than the maximum number of section ("
-                            + maximumSectionCount + ")");
+        if (currentSectionCount == null) {
+            throwExceptionOrLogError(validable, "the current number of section is undefined", throwException, reporter);
+            validationLevel = ValidationLevel.min(validationLevel, ValidationLevel.STATE_ESTIMATION);
+        } else {
+            if (currentSectionCount < 0) {
+                throwExceptionOrLogError(validable, "the current number of section (" + currentSectionCount
+                        + ") should be greater than or equal to 0", throwException, reporter);
+                validationLevel = ValidationLevel.min(validationLevel, ValidationLevel.STATE_ESTIMATION);
+            }
+            if (currentSectionCount > maximumSectionCount) {
```

Comment:
Same remark: should be forbidden at all. Only a transitory state is allowed where the current step is not defined.